### PR TITLE
Create user home resource

### PR DESCRIPTION
**Pull request by dhoer**
Creates and populates home directory without requiring logon.  Inspiration taken from https://github.com/opscode-cookbooks/omnibus/blob/master/recipes/_user.rb#L48-62.

### CODE REVIEW

**Comment by dhoer** [comment]
Is there interest in adding this feature to windows cookbook, or should does it belong somewhere else?

**Comment by jdmundrawala** [comment]
Looks interesting, I think it would be a nice addition

**Comment by dhoer** [comment]
Awesome! When do you think it would be released?  I have it as a dependency in windows_desktop::headless_display cookbook: https://github.com/dhoer/chef-windows_desktop/blob/master/recipes/headless_display.rb#L27-30

File: providers/home.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  task_name = "create_#{@new_resource.name}_home"
+  user_home = ::File.join(ENV['SYSTEMDRIVE'], 'Users', @new_resource.name)
+    .gsub(::File::SEPARATOR, ::File::ALT_SEPARATOR)
+
+  # Run whoami as scheduled task to get LocalServer group privilege:
+  # SeAssignPrimaryTokenPrivilege (Replace a process-level token)
+  if task_query(task_name).empty?
+    cmd =  "schtasks /Create /TN \"#{task_name}\" "
+    cmd += "/SC once /SD \"01/01/2003\" /ST \"00:00\" /TR \"whoami.exe\" "
+    cmd += "/RU \"#{@new_resource.name}\" /RP \"#{@new_resource.password}\" "
```
Comment (jdmundrawala) [inline providers/home.rb]: won't this password be available for all to see?

Comment (dhoer) [inline providers/home.rb]: You are correct. Good catch.  I can delete this task after it completes.  Or is there a better way to run whoami with SeAssignPrimaryTokenPrivilege?

Comment (dhoer) [inline providers/home.rb]: Wait, I don't know if this password is visible or not. I can't tell from the documentation: https://msdn.microsoft.com/en-us/library/windows/desktop/bb736357%28v=vs.85%29.aspx.  It still would make sense to delete the task to be on the safe side.

Comment (dhoer) [inline providers/home.rb]: I can try to run execute resource with sensitive turned on, since I don't see that option for shell_out.

Comment (jdmundrawala) [inline providers/home.rb]: As far as I know, all that does is prevent chef from reporting the sensitive information. I'm not really sure what the right thing to do here is. I'll ping people, see what they think

**Comment by dhoer** [comment]
@jdmundrawala What do you think of this solution?

**Comment by dhoer** [comment]
@jdmundrawala  Thanks for the code review and making this better.  I'm closing this pull request.  You can still add it in the future, if you want.  This home feature is now available in windows_desktop cookbook.

**Pull request closed without merging**